diff -up gst-ffmpeg-0.10.13/ext/libswscale/gstffmpegscale.c.orc gst-ffmpeg-0.10.13/ext/libswscale/gstffmpegscale.c
--- gst-ffmpeg-0.10.13/ext/libswscale/gstffmpegscale.c.orc	2019-11-20 10:08:09.730167627 +0100
+++ gst-ffmpeg-0.10.13/ext/libswscale/gstffmpegscale.c	2019-11-20 10:13:36.888577009 +0100
@@ -636,7 +636,7 @@ gst_ffmpegscale_set_caps (GstBaseTransfo
   swsflags = (mmx_flags & ORC_TARGET_MMX_MMX ? SWS_CPU_CAPS_MMX : 0)
       | (mmx_flags & ORC_TARGET_MMX_MMXEXT ? SWS_CPU_CAPS_MMX2 : 0)
       | (mmx_flags & ORC_TARGET_MMX_3DNOW ? SWS_CPU_CAPS_3DNOW : 0)
-      | (altivec_flags & ORC_TARGET_ALTIVEC_ALTIVEC ? SWS_CPU_CAPS_ALTIVEC : 0);
+      | (altivec_flags & ORC_TARGET_POWERPC_ALTIVEC ? SWS_CPU_CAPS_ALTIVEC : 0);
 #else
   mmx_flags = 0;
   altivec_flags = 0;
diff -up gst-ffmpeg-0.10.13/ext/libpostproc/gstpostproc.c.orc gst-ffmpeg-0.10.13/ext/libpostproc/gstpostproc.c
--- gst-ffmpeg-0.10.13/ext/libpostproc/gstpostproc.c.orc	2019-11-20 10:23:40.014758297 +0100
+++ gst-ffmpeg-0.10.13/ext/libpostproc/gstpostproc.c	2019-11-20 10:27:56.561215105 +0100
@@ -299,7 +299,7 @@ change_context (GstPostProc * postproc,
     ppflags = (mmx_flags & ORC_TARGET_MMX_MMX ? PP_CPU_CAPS_MMX : 0)
         | (mmx_flags & ORC_TARGET_MMX_MMXEXT ? PP_CPU_CAPS_MMX2 : 0)
         | (mmx_flags & ORC_TARGET_MMX_3DNOW ? PP_CPU_CAPS_3DNOW : 0)
-        | (altivec_flags & ORC_TARGET_ALTIVEC_ALTIVEC ? PP_CPU_CAPS_ALTIVEC :
+        | (altivec_flags & ORC_TARGET_POWERPC_ALTIVEC ? PP_CPU_CAPS_ALTIVEC :
         0);
 #else
     mmx_flags = 0;
