--- gst-ffmpeg-0.10.2/ext/ffmpeg/gstffmpegdemux.c.syslibs	2006-10-25 19:20:25.000000000 +0200
+++ gst-ffmpeg-0.10.2/ext/ffmpeg/gstffmpegdemux.c	2007-01-19 16:57:04.000000000 +0100
@@ -131,9 +131,6 @@ gst_ffmpegdemux_averror (gint av_errno)
   const gchar *message = NULL;
 
   switch (av_errno) {
-    case AVERROR_UNKNOWN:
-      message = "Unknown error";
-      break;
     case AVERROR_IO:
       message = "Input/output error";
       break;
--- gst-ffmpeg-0.10.2/ext/ffmpeg/gstffmpegcodecmap.c~	2007-09-21 10:10:51.000000000 +0200
+++ gst-ffmpeg-0.10.2/ext/ffmpeg/gstffmpegcodecmap.c	2007-09-21 10:10:51.000000000 +0200
@@ -523,7 +523,6 @@ gst_ffmpeg_codecid_to_caps (enum CodecID
       caps = gst_ff_vid_caps_new (context, "video/x-theora", NULL);
       break;
 
-    case CODEC_ID_AAC:
     case CODEC_ID_MPEG4AAC:
       caps = gst_ff_aud_caps_new (context, "audio/mpeg",
           "mpegversion", G_TYPE_INT, 4, NULL);
@@ -2492,7 +2491,6 @@ gst_ffmpeg_get_codecid_longname (enum Co
     case CODEC_ID_THEORA:
       name = "Theora video";
       break;
-    case CODEC_ID_AAC:
     case CODEC_ID_MPEG4AAC:
       name = "MPEG-2/4 AAC audio";
       break;
@@ -3162,6 +3162,7 @@ gst_ffmpeg_avpicture_fill (AVPicture * p
   return 0;
 }
 
+#if 0
 /**
  * Convert image 'src' to 'dst'.
  *
@@ -3196,3 +3197,4 @@ gst_ffmpeg_img_convert (AVPicture * dst,
   return img_convert (dst, dst_pix_fmt, src, src_pix_fmt, src_width,
       src_height);
 }
+#endif
--- gst-ffmpeg-0.10.2/ext/ffmpeg/gstffmpegcfg.c~	2007-10-10 15:16:29.000000000 +0200
+++ gst-ffmpeg-0.10.2/ext/ffmpeg/gstffmpegcfg.c	2007-10-10 15:16:29.000000000 +0200
@@ -242,6 +242,7 @@ gst_ffmpeg_pred_method_get_type (void)
       {FF_PRED_LEFT, "Left", "left"},
       {FF_PRED_PLANE, "Plane", "plane"},
       {FF_PRED_MEDIAN, "Median", "median"},
+      { 0, NULL, NULL }
     };
 
     ffmpeg_pred_method =
--- gst-ffmpeg-0.10.3/ext/ffmpeg/gstffmpegdec.c~	2007-12-15 11:49:31.000000000 +0100
+++ gst-ffmpeg-0.10.3/ext/ffmpeg/gstffmpegdec.c	2007-12-15 11:49:31.000000000 +0100
@@ -488,6 +488,7 @@ gst_ffmpegdec_open (GstFFMpegDec * ffmpe
 
   ffmpegdec->opened = TRUE;
   ffmpegdec->is_realvideo = FALSE;
+  ffmpegdec->context->codec_type = oclass->in_plugin->type;
 
   GST_LOG_OBJECT (ffmpegdec, "Opened ffmpeg codec %s, id %d",
       oclass->in_plugin->name, oclass->in_plugin->id);
@@ -1267,11 +1267,7 @@ get_output_buffer (GstFFMpegDec * ffmpeg
     gst_ffmpeg_avpicture_fill (&pic, GST_BUFFER_DATA (*outbuf),
         ffmpegdec->context->pix_fmt, width, height);
 
-    /* the original convert function did not do the right thing, this
-     * is a patched up version that adjust widht/height so that the
-     * ffmpeg one works correctly. */
-    gst_ffmpeg_img_convert (&pic, ffmpegdec->context->pix_fmt,
-        (AVPicture *) ffmpegdec->picture,
+    av_picture_copy(&pic, (AVPicture *) ffmpegdec->picture,
         ffmpegdec->context->pix_fmt, width, height);
   }
   return ret;
diff -up gst-ffmpeg-0.10.3/ext/ffmpeg/Makefile.in~ gst-ffmpeg-0.10.3/ext/ffmpeg/Makefile.in
--- gst-ffmpeg-0.10.3/ext/ffmpeg/Makefile.in~	2008-02-10 22:00:45.000000000 +0100
+++ gst-ffmpeg-0.10.3/ext/ffmpeg/Makefile.in	2008-02-10 22:00:45.000000000 +0100
@@ -78,7 +78,7 @@ am_libgstffmpeg_la_OBJECTS = libgstffmpe
 	libgstffmpeg_la-gstffmpegdemux.lo \
 	libgstffmpeg_la-gstffmpegmux.lo \
 	libgstffmpeg_la-gstffmpegdeinterlace.lo \
-	libgstffmpeg_la-gstffmpegscale.lo
+
 libgstffmpeg_la_OBJECTS = $(am_libgstffmpeg_la_OBJECTS)
 libgstffmpeg_la_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(libgstffmpeg_la_CFLAGS) \
@@ -271,7 +271,6 @@ libgstffmpeg_la_SOURCES = gstffmpeg.c	\
 			  gstffmpegdemux.c	\
 			  gstffmpegmux.c    \
 			  gstffmpegdeinterlace.c	\
-			  gstffmpegscale.c
 
 libgstffmpeg_la_CFLAGS = $(GST_CFLAGS) $(GST_PLUGINS_BASE_CFLAGS) \
 	$(FFMPEG_CFLAGS)
@@ -363,7 +362,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgstffmpeg_la-gstffmpegenc.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgstffmpeg_la-gstffmpegmux.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgstffmpeg_la-gstffmpegprotocol.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libgstffmpeg_la-gstffmpegscale.Plo@am__quote@
 
 .c.o:
 @am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
@@ -449,13 +447,6 @@ libgstffmpeg_la-gstffmpegdeinterlace.lo:
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libgstffmpeg_la_CFLAGS) $(CFLAGS) -c -o libgstffmpeg_la-gstffmpegdeinterlace.lo `test -f 'gstffmpegdeinterlace.c' || echo '$(srcdir)/'`gstffmpegdeinterlace.c
 
-libgstffmpeg_la-gstffmpegscale.lo: gstffmpegscale.c
-@am__fastdepCC_TRUE@	$(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libgstffmpeg_la_CFLAGS) $(CFLAGS) -MT libgstffmpeg_la-gstffmpegscale.lo -MD -MP -MF $(DEPDIR)/libgstffmpeg_la-gstffmpegscale.Tpo -c -o libgstffmpeg_la-gstffmpegscale.lo `test -f 'gstffmpegscale.c' || echo '$(srcdir)/'`gstffmpegscale.c
-@am__fastdepCC_TRUE@	mv -f $(DEPDIR)/libgstffmpeg_la-gstffmpegscale.Tpo $(DEPDIR)/libgstffmpeg_la-gstffmpegscale.Plo
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='gstffmpegscale.c' object='libgstffmpeg_la-gstffmpegscale.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libgstffmpeg_la_CFLAGS) $(CFLAGS) -c -o libgstffmpeg_la-gstffmpegscale.lo `test -f 'gstffmpegscale.c' || echo '$(srcdir)/'`gstffmpegscale.c
-
 mostlyclean-libtool:
 	-rm -f *.lo
 
diff -up gst-ffmpeg-0.10.3/ext/ffmpeg/gstffmpeg.c~ gst-ffmpeg-0.10.3/ext/ffmpeg/gstffmpeg.c
--- gst-ffmpeg-0.10.3/ext/ffmpeg/gstffmpeg.c~	2008-02-10 21:59:20.000000000 +0100
+++ gst-ffmpeg-0.10.3/ext/ffmpeg/gstffmpeg.c	2008-02-10 21:59:20.000000000 +0100
@@ -143,8 +143,8 @@ plugin_init (GstPlugin * plugin)
   gst_ffmpegdemux_register (plugin);
   gst_ffmpegmux_register (plugin);
   gst_ffmpegdeinterlace_register (plugin);
-  gst_ffmpegscale_register (plugin);
 #if 0
+  gst_ffmpegscale_register (plugin);
   gst_ffmpegcsp_register (plugin);
 #endif
 
diff -up gst-ffmpeg-0.10.3/ext/ffmpeg/Makefile.am~ gst-ffmpeg-0.10.3/ext/ffmpeg/Makefile.am
--- gst-ffmpeg-0.10.3/ext/ffmpeg/Makefile.am~	2008-02-10 22:00:50.000000000 +0100
+++ gst-ffmpeg-0.10.3/ext/ffmpeg/Makefile.am	2008-02-10 22:00:50.000000000 +0100
@@ -9,7 +9,6 @@ libgstffmpeg_la_SOURCES = gstffmpeg.c	\
 			  gstffmpegdemux.c	\
 			  gstffmpegmux.c    \
 			  gstffmpegdeinterlace.c	\
-			  gstffmpegscale.c
 
 libgstffmpeg_la_CFLAGS = $(GST_CFLAGS) $(GST_PLUGINS_BASE_CFLAGS) \
 	$(FFMPEG_CFLAGS)
--- gst-ffmpeg-0.10.3/ext/ffmpeg/gstffmpegmux.c.syslibs	2006-10-25 19:20:25.000000000 +0200
+++ gst-ffmpeg-0.10.3/ext/ffmpeg/gstffmpegmux.c	2007-01-19 16:57:04.000000000 +0100
@@ -481,7 +481,7 @@ gst_ffmpegmux_collected (GstCollectPads 
     ffmpegmux->opened = TRUE;
 
     /* flush the header so it will be used as streamheader */
-    put_flush_packet (&ffmpegmux->context->pb);
+    put_flush_packet (ffmpegmux->context->pb);
   }
 
   /* take the one with earliest timestamp,
@@ -556,8 +556,8 @@ gst_ffmpegmux_collected (GstCollectPads 
     /* close down */
     av_write_trailer (ffmpegmux->context);
     ffmpegmux->opened = FALSE;
-    put_flush_packet (&ffmpegmux->context->pb);
-    url_fclose (&ffmpegmux->context->pb);
+    put_flush_packet (ffmpegmux->context->pb);
+    url_fclose (ffmpegmux->context->pb);
     gst_pad_push_event (ffmpegmux->srcpad, gst_event_new_eos ());
     return GST_FLOW_UNEXPECTED;
   }
@@ -598,7 +598,7 @@ gst_ffmpegmux_change_state (GstElement *
       }
       if (ffmpegmux->opened) {
         ffmpegmux->opened = FALSE;
-        url_fclose (&ffmpegmux->context->pb);
+        url_fclose (ffmpegmux->context->pb);
       }
       break;
     case GST_STATE_CHANGE_READY_TO_NULL:
