--- gst-ffmpeg-0.10.2/ext/ffmpeg/gstffmpegdemux.c.syslibs	2006-10-25 19:20:25.000000000 +0200
+++ gst-ffmpeg-0.10.2/ext/ffmpeg/gstffmpegdemux.c	2007-01-19 16:57:04.000000000 +0100
@@ -131,9 +131,6 @@ gst_ffmpegdemux_averror (gint av_errno)
   const gchar *message = NULL;
 
   switch (av_errno) {
-    case AVERROR_UNKNOWN:
-      message = "Unknown error";
-      break;
     case AVERROR_IO:
       message = "Input/output error";
       break;
--- gst-ffmpeg-0.10.2/ext/ffmpeg/gstffmpegcodecmap.c~	2007-09-21 10:10:51.000000000 +0200
+++ gst-ffmpeg-0.10.2/ext/ffmpeg/gstffmpegcodecmap.c	2007-09-21 10:10:51.000000000 +0200
@@ -523,7 +523,6 @@ gst_ffmpeg_codecid_to_caps (enum CodecID
       caps = gst_ff_vid_caps_new (context, "video/x-theora", NULL);
       break;
 
-    case CODEC_ID_AAC:
     case CODEC_ID_MPEG4AAC:
       caps = gst_ff_aud_caps_new (context, "audio/mpeg",
           "mpegversion", G_TYPE_INT, 4, NULL);
@@ -2492,7 +2491,6 @@ gst_ffmpeg_get_codecid_longname (enum Co
     case CODEC_ID_THEORA:
       name = "Theora video";
       break;
-    case CODEC_ID_AAC:
     case CODEC_ID_MPEG4AAC:
       name = "MPEG-2/4 AAC audio";
       break;
--- gst-ffmpeg-0.10.2/ext/ffmpeg/gstffmpegcfg.c~	2007-10-10 15:16:29.000000000 +0200
+++ gst-ffmpeg-0.10.2/ext/ffmpeg/gstffmpegcfg.c	2007-10-10 15:16:29.000000000 +0200
@@ -242,6 +242,7 @@ gst_ffmpeg_pred_method_get_type (void)
       {FF_PRED_LEFT, "Left", "left"},
       {FF_PRED_PLANE, "Plane", "plane"},
       {FF_PRED_MEDIAN, "Median", "median"},
+      { 0, NULL, NULL }
     };
 
     ffmpeg_pred_method =
--- gst-ffmpeg-0.10.3/ext/ffmpeg/gstffmpegdec.c~	2007-12-15 11:49:31.000000000 +0100
+++ gst-ffmpeg-0.10.3/ext/ffmpeg/gstffmpegdec.c	2007-12-15 11:49:31.000000000 +0100
@@ -488,6 +488,7 @@ gst_ffmpegdec_open (GstFFMpegDec * ffmpe
 
   ffmpegdec->opened = TRUE;
   ffmpegdec->is_realvideo = FALSE;
+  ffmpegdec->context->codec_type = oclass->in_plugin->type;
 
   GST_LOG_OBJECT (ffmpegdec, "Opened ffmpeg codec %s, id %d",
       oclass->in_plugin->name, oclass->in_plugin->id);
